packages/calcite-components/src/utils/dom.ts

```diff
@@ -419,6 +419,17 @@ export function filterDirectChildren<T extends Element>(el: Element, selector: s
   return Array.from(el.children).filter((child): child is T => child.matches(selector));
 }
 
+/**
+ * Filters an array of HTML elements by the provided css selector string.
+ *
+ * @param {Element[]} elements An array of elements, such as one returned by HTMLSlotElement.assignedNodes().
```

Comment:
Should this say `HTMLSlotElement.assignedNodes().` or `HTMLSlotElement.assignedElements().`? If it should be nodes, I don't think `.matches` would work for TextContent nodes.